### ISSUE TITLE
Segfault when deleting mesh

### ISSUE THREAD

**Issue by nh2** [cnr-isti-vclab/meshlab#1240]
* Meshlab 2021.10 on NixOS 21.05
* When hiding some meshes and then using Rightclick -> `Delete current mesh` a couple times:

```c
Thread 1 ".meshlab-wrappe" received signal SIGSEGV, Segmentation fault.
0x000000000046b7fc in GLArea::displayInfo(QPainter*) ()
(gdb) bt
#0  0x000000000046b7fc in GLArea::displayInfo(QPainter*) ()
#1  0x0000000000472ca0 in GLArea::paintEvent(QPaintEvent*) ()
#2  0x00007ffff77579ce in QWidget::event(QEvent*) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Widgets.so.5
#3  0x00007ffff771576f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Widgets.so.5
#4  0x00007ffff7f9d94a in MeshLabApplication::notify(QObject*, QEvent*) () from /nix/store/gvmv2jvx443lww1cg6m0lcymql95g5vp-meshlab-2021.10/lib/meshlab/libmeshlab-common.so
#5  0x00007ffff6bf340a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Core.so.5
#6  0x00007ffff774f926 in QWidgetPrivate::sendPaintEvent(QRegion const&) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Widgets.so.5
#7  0x00007ffff775016b in QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) ()
   from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Widgets.so.5
#8  0x00007ffff7750ec2 in QWidgetPrivate::paintOnScreen(QRegion const&) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Widgets.so.5
#9  0x00007ffff7750ff4 in QWidgetPrivate::syncBackingStore() () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Widgets.so.5
#10 0x00007ffff775812d in QWidget::event(QEvent*) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Widgets.so.5
#11 0x00007ffff771576f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Widgets.so.5
#12 0x00007ffff7f9d94a in MeshLabApplication::notify(QObject*, QEvent*) () from /nix/store/gvmv2jvx443lww1cg6m0lcymql95g5vp-meshlab-2021.10/lib/meshlab/libmeshlab-common.so
#13 0x00007ffff6bf340a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Core.so.5
#14 0x00007ffff6bf6451 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Core.so.5
#15 0x00007ffff6c4bd93 in ?? () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Core.so.5
#16 0x00007ffff3ac9aab in g_main_dispatch (context=0x7fffe0005000) at ../glib/gmain.c:3381
#17 g_main_context_dispatch (context=0x7fffe0005000) at ../glib/gmain.c:4099
#18 0x00007ffff3ac9d58 in g_main_context_iterate (context=context@entry=0x7fffe0005000, block=block@entry=1, dispatch=dispatch@entry=1, self=<optimized out>) at ../glib/gmain.c:4175
#19 0x00007ffff3ac9e0f in g_main_context_iteration (context=0x7fffe0005000, may_block=1) at ../glib/gmain.c:4240
#20 0x00007ffff6c4b3f0 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Core.so.5
#21 0x00007ffff6bf1dab in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Core.so.5
#22 0x00007ffff6bfa3a0 in QCoreApplication::exec() () from /nix/store/gvzqkcdmm5yd0jxkr9jka4cfm21kvbdl-qtbase-5.15.3/lib/libQt5Core.so.5
#23 0x000000000044eabf in main ()
```

I haven't had the time yet to try on the latest version, but maybe it's easily reproducible.

**Comment by alemuntoni** [comment]
It has been already solved [here](b5a8e5138508dc76c8072a8e37c52894c2fe90d0). Let me know if you still can reproduce the issue in the last release.

**Comment by nh2** [comment]
Ah great, thank you! https://github.com/cnr-isti-vclab/meshlab/commit/b5a8e5138508dc76c8072a8e37c52894c2fe90d0
